src/actions/AppActions.ts

```diff
@@ -314,40 +314,54 @@ export async function runTask(
   }
 }
 
-export async function deploy(contractName: string) {
+async function deploy(contractName: string) {
   const mainFileName = "out/main.wasm";
   const projectModel = appStore.getProject().getModel();
-  pushStatus("Deploying Contract");
   await Service.deployContract(contractName, projectModel.getFile(mainFileName), this);
-  popStatus();
   projectModel.getFile(mainFileName);
 }
 
+async function createAccountForContract(contractName: string) {
+  // TODO: Remove ugly hack with window
+  const app = (window as any).app;
+  const keyPair = await app.state.keyStore.getKey(contractName);
+  // if no keypair in keystore, it means the account does not exist.
+  // maybe there is a better way to check this?
+  if (!keyPair.getPublicKey()) {
```

Comment:
I think we can remove the code that creates the throw away temp account now, because it's not needed for anything. I was going to do it sometime this week, but if you want to do it and get to it first that's cool too.

Reply:
@janedegtiareva do you mean that this branch never gets executed?